### PR TITLE
DPO integration test - logits comparision against Pytorch

### PR DESCRIPTION
**Pull request by igorts-git**
# Description

Add a logits comparison test for MaxText DPO vs HuggingFace TRL implementation.

The new test consists of two parts:
1. `generate_dpo_golden_data_and_compare_pytorch_logits.py` - compares DPO in MaxText vs HF TRL and saves golden data json file.
2. A unit test that only runs MaxText DPO and compares the results against the golden data in the json files.

This separation allows us to not have a dependency on TRL in CI.
The tests are running on CPU to reduce the GPU/TPU floating point differences. We use a tiny 2-layer qwen2-like model to make sure that this unit test runs fast.

A separate PR will be sent out to add a similar test for ORPO. The Tunix implementation would first need to be adjusted a little bit to match the canonical implementation (see https://github.com/google/tunix/pull/1574).

# Tests

Ran the test manually.

# Checklist

Before submitting this PR, please make sure (put X in square brackets):
- [X] I have performed a self-review of my code. For an optional AI review, add the `gemini-review` label.
- [X] I have necessary comments in my code, particularly in hard-to-understand areas.
- [X] I have run end-to-end tests tests and provided workload links above if applicable.
- [X] I have made or will make corresponding changes to the doc if needed, including adding new documentation pages to the relevant Table of Contents (toctree directive) as explained in [our documentation](https://maxtext.readthedocs.io/en/latest/development.html#adding-new-documentation-files).

### CODE REVIEW

File: tests/post_training/integration/dpo_correctness_base.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+
+    self.captured_metrics.append(
+        {
+            "loss": float(train_loss),
+            "accuracy": accuracy,
+            "margin": margin,
+            "chosen_logps": chosen_logps,
+            "rejected_logps": rejected_logps,
+        }
+    )
```
[the remark this answers is not in the document]
Comment (igorts-git) [reply tests/post_training/integration/dpo_correctness_base.py]: done

File: tests/assets/logits_generation/dpo_pytorch_helpers.py [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+      ]
+  )
+
+  training_args = DPOConfig(
+      output_dir="/tmp/trl_ref",
+      beta=beta,
+      max_length=256,
+      use_cpu=True,
+      remove_unused_columns=False,
+  )
```
[the remark this answers is not in the document]
Comment (igorts-git) [reply tests/assets/logits_generation/dpo_pytorch_helpers.py]: done

File: tests/post_training/integration/dpo_correctness_base.py [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+      data_columns: list[str],
+      max_prompt_len: int | None = None,
+      extra_args: list[str] | None = None,
+  ) -> Any:
+    """Helper to build MaxText JAX config object for DPO."""
+    dataset_path = os.path.abspath(f"tests/assets/local_datasets/dpo/{dataset_filename}")
+    argv = [
+        "src/maxtext/configs/base.yml",
+        "model_name=qwen2.5-1.5b",
+        f"tokenizer_path={model_id}",
```
[the remark this answers is not in the document]
Comment (igorts-git) [reply tests/post_training/integration/dpo_correctness_base.py]: done

**Review by khatwanimohit — APPROVED** [review]

**Review by richjames0 — APPROVED** [review]
lgtm
